src/state/cached_state.rs

```diff
@@ -82,16 +81,19 @@ impl<T: StateReader> CachedState<T> {
 }
 
 impl<T: StateReader> StateReader for CachedState<T> {
-    fn get_class_hash_at(&mut self, contract_address: &Address) -> Result<ClassHash, StateError> {
+    fn get_class_hash_at(&self, contract_address: &Address) -> Result<ClassHash, StateError> {
         if self.cache.get_class_hash(contract_address).is_none() {
-            let class_hash = match self.state_reader.get_class_hash_at(contract_address) {
-                Ok(x) => x,
-                Err(StateError::NoneContractState(_)) => [0; 32],
-                Err(e) => return Err(e),
-            };
-            self.cache
-                .class_hash_initial_values
-                .insert(contract_address.clone(), class_hash);
+            match self.state_reader.get_class_hash_at(contract_address) {
+                Ok(x) => {
+                    return Ok(x);
+                }
+                Err(StateError::NoneContractState(_)) => {
+                    return Ok([0; 32]);
+                }
+                Err(e) => {
+                    return Err(e);
```

Comment:
returns should not be necessary if the match statement is at the end of the function

Reply:
It is not at the end of the function